include/seastar/util/log.hh

```diff
@@ -120,6 +123,53 @@ public:
 
     /// \cond internal
     /// \brief used to hold the log format string and the caller's source_location.
+#ifdef SEASTAR_LOGGER_COMPILE_TIME_FMT
+    template<typename... Args>
+    struct format_info {
+        /// implicitly construct format_info from a constant format string
+        /// \param fmt - {fmt} style format string
+        template<
+            typename S,
+            std::enable_if_t<std::is_convertible_v<const S&, std::string_view>, int> = 0>
+        FMT_CONSTEVAL inline format_info(const S& format,
+                           compat::source_location loc = compat::source_location::current()) noexcept
+            : format(format)
+            , loc(loc)
+        {}
```

Comment:
I don't understand this constructor. If S can be converted into format_string<Args...>, why wouldn't the second constructor suffice?

Reply:
i think the reason why the second ctor does not suffice is: C++ does not perform implicit conversion to a single variable to the expected type multiple times.

see https://godbolt.org/z/h65fzhMEx for a reproducer deduced from our real-world use case. if 
```c++
#define CXX_MULTI_IMPLICIT_CONVERSION false
```
is changed to 
```c++
#define CXX_MULTI_IMPLICIT_CONVERSION true
```
the compilers would fail to compile the source. 

and the C++ standard explicitly forbids this. see https://eel.is/c++draft/class.conv.general#4. i am quoting it here 

> At most one user-defined conversion (constructor or conversion function) is implicitly applied to a single value[.](https://eel.is/c++draft/class.conv.general#4.sentence-1)

but it comes with a smaller code list to demonstrate the use case. so i think it's also a good reading.